src/sentry/workflow_engine/models/incident_groupopenperiod.py

```diff
@@ -127,70 +178,104 @@ def create_relationship(self, incident, open_period):
-            detector_id = AlertRuleDetector.objects.get(alert_rule_id=alert_rule.id).detector_id
-            pending_open_periods = GroupOpenPeriod.objects.filter(
-                data__pending_incident_detector_id=detector_id,
-                group__project__in=list(incident.projects.all()),
-            )
+    except IncidentGroupOpenPeriod.DoesNotExist:
+        logger.warning(
+            "No IncidentGroupOpenPeriod relationship found",
+            extra={
+                "open_period_id": open_period.id,
+            },
+        )
+        return
 
-            for open_period in pending_open_periods:
-                # Create the relationship
-                relationship = self.create_relationship(incident, open_period)
-                if relationship:
-                    # Remove the pending flag from the open_period data
-                    data = open_period.data or {}
-                    data.pop("pending_incident_detector_id", None)
-                    open_period.update(data=data)
+    severity = (
+        IncidentStatus.CRITICAL if priority == PriorityLevel.HIGH else IncidentStatus.WARNING
+    )  # this assumes that LOW isn't used for metric issues
 
-        except Exception as e:
-            logger.exception(
-                "Failed to create pending IncidentGroupOpenPeriod relationships",
-                extra={
-                    "incident_id": incident.id,
-                    "alert_rule_id": alert_rule.id,
-                    "error": str(e),
-                },
+    update_incident_status(
+        incident,
+        severity,
+        status_method=IncidentStatusMethod.RULE_TRIGGERED,
+    )
+
+
+def dual_update_incident_and_open_period(
+    group: Group,
+    new_status: int,
+) -> None:
+    from sentry.incidents.logic import update_incident_status
+    from sentry.incidents.models.incident import Incident, IncidentStatus, IncidentStatusMethod
+    from sentry.incidents.utils.process_update_helpers import calculate_event_date_from_update_date
+
+    open_period = get_latest_open_period(group)
+    if open_period is None:
+        logger.warning("No open period found for group", extra={"group_id": group.id})
+        return
+
+    if not features.has(
+        "organizations:workflow-engine-single-process-metric-issues", group.project.organization
+    ):
+        return
+
+    # get the incident for the open period
+    try:
+        incident_id = IncidentGroupOpenPeriod.objects.get(group_open_period=open_period).incident_id
+        incident = Incident.objects.get(id=incident_id)
+
+    except IncidentGroupOpenPeriod.DoesNotExist:
+        logger.warning(
+            "No IncidentGroupOpenPeriod relationship found",
+            extra={
+                "open_period_id": open_period.id,
+            },
+        )
+        return
+
+    subscription = QuerySubscription.objects.get(id=int(incident.subscription_id))
+    snuba_query = SnubaQuery.objects.get(id=subscription.snuba_query_id)
```

Comment:
might be worth simplifying this down to a single query that uses `prefetch_related` to grab the associated snuba query. might even be able to filter down to just getting the timestamp to further improve query performance.

something like:
```
subscriptions = QuerySubscription.objects.filter(id=int(incident.subscription_id)).prefetch_related("snuba_query")
subscription = subscriptions.first()
```
will get it down to a single query, but might be able to further reduce the data being fetched for the snuba query.